src/include/coredumpctl.h

```diff
@@ -37,3 +40,9 @@ struct PrCtl {
 #else
 struct PrCtl {};
 #endif
+
+// usually this structure is definded, but not referenced.
+// Which will generate unused-variable if not refed with (void)
+#define UNSET_DUMPABLE PrCtl unset_dumpable; (void)unset_dumpable
```

Comment:
the reason that you have
```
/home/jenkins/workspace/ceph-master/src/test/bufferlist.cc:490:11:
warning: unused variable 'unset_dumpable' [-Wunused-variable]
    PrCtl unset_dumpable;
          ^
```

is that `PrCtl` an pretty much an empty struct on FreeBSD. so we could just have following change
```diff
diff --git a/src/include/coredumpctl.h b/src/include/coredumpctl.h
index e4424941f4..c3e1109f89 100644
--- a/src/include/coredumpctl.h
+++ b/src/include/coredumpctl.h
@@ -1,3 +1,5 @@
+#pragma once
+
 #ifdef HAVE_SYS_PRCTL_H
 #include <iostream>
 #include <sys/prctl.h>
@@ -35,5 +37,8 @@ struct PrCtl {
   }
 };
 #else
-struct PrCtl {};
+struct PrCtl {
+  // to silence the Wunused-variable warning
+  PrCtl() {}
+};
 #endif
```